16_microbiome_community.Rmd

```diff
@@ -49,24 +49,222 @@ Community composition can be visualized with heatmap where other axis represent
+
+# Prints taxa and their clusters
+taxa_clusters 
+```
+
+```{r pheatmap4}
+# Creates clusters, and adds information to rowData
+rowData(tse_phylum_subset)$clusters <- cutree(tree = taxa_dendrogram, k = 3)
+
+# Prints taxa and their clusters
+rowData(tse_phylum_subset)$clusters
+```
+
+Now we can create heatmap with additional annotations.
+
+```{r pheatmap5}
+# Creates data frame that includes cluster data
+taxa_clusters <- rowData(tse_phylum_subset)$clusters
+taxa_clusters <- as.character(taxa_clusters)
+taxa_clusters <- data.frame(cluster = taxa_clusters)
+row.names(taxa_clusters) <- rownames(tse_phylum_subset)
+
+# Creates data frame that includes sample type data
+sample_types <- unfactor(colData(tse_phylum_subset)$SampleType)
+sample_types <- data.frame(sample_types = sample_types)
+row.names(sample_types) <- colnames(tse_phylum_subset)
+
+pheatmap(mat, annotation_row = taxa_clusters, 
+         annotation_col = sample_types)
+```
+
+In addition to _pheatmap_ package, there are also other packages that provide 
+functions for more complex heatmaps. One example is _iheatmapr_ package. Examples
+of using it you can find from its 
+[vignette](https://docs.ropensci.org/iheatmapr/articles/full_vignettes/iheatmapr.html).
+
+# Cross-correlation
+
+With cross-correlation analysis, we can analyze how strongly and how different
+variables are associated between each other. For instance, we can analyze if 
+higher presence of specific taxon equals to higher level of biomolecule. 
+
+Here, we analyze associations between taxa correlate and lipids. Data is from 
+following publication Lahti _et al_. (2015) [Associations between the human intestinal 
+microbiota, Lactobacillus rhamnosus GG and serum lipids indicated by 
+integrated analysis of high-throughput profiling 
+data](https://peerj.com/articles/32/).
+
+```{r cross-correlation1}
+# Imports the data
+tse <- microbiomeDataSets::peerj32()
+
+# Microbiome data
+tse[[1]] 
+```
+
+```{r cross-correlation2}
+# Lipid data
+tse[[2]]
+```
+
+```{r cross-correlation3}
+if(!require(microbiome)){
+    BiocManager::install("microbiome")
+}
+
+# Does log10 transform for microbiome data
+tse[[1]] <- transformSamples(tse[[1]], method = "log10", pseudocount = 1)
+
+# Gets microbiome and lipid data to cross-correlate
+x <- t(assay(tse[[1]], "log10"))
+y <- t(assay(tse[[2]], "counts"))
+
+# Cross correlates data sets
+correlation_table <- microbiome::associate(x, y, method = "spearman", mode = "table", 
+                                           p.adj.threshold = 0.05, n.signif = 1)
+
+knitr::kable(head(correlation_table))
+```
+
+Manipulates and reorders the table
+
+```{r cross-correlation4}
+if(!require(reshape2)){
+    install.packages("reshape2")
+}
+# Gets taxa that has at least one statistically significant correlation
+taxa <- as.character(
+  unique(correlation_table[correlation_table[["p.adj"]] < 0.05 &
+                             abs(correlation_table[["Correlation"]]) > 0, "X1"]))
```

Comment:
see below for alternative filtering command with pipes

Reply:
This is easier to understand, changed